polkadot/node/network/dispute-distribution/src/sender/send_task.rs

```diff
@@ -153,6 +169,7 @@ impl<M: 'static + Send + Sync> SendTask<M> {
 			"Cleaning up deliveries"
 		);
 		self.deliveries.retain(|k, _| new_authorities.contains(k));
+		self.rejected_deliveries.retain(|k, _| new_authorities.contains(k));
```

Comment:
Can this and deliveries be merged?

Reply:
I'm keeping track of the rejected deliveries separately because they are removed from `delivereis` [here](https://github.com/paritytech/polkadot-sdk/blob/b3542e6602d749218ad341f8e05e13693c599880/polkadot/node/network/dispute-distribution/src/sender/send_task.rs#L255).

We can merge them of course but in that case we have to modify `DeliveryStatus` to keep track of rejected and failed requests and it will complicate the code a tad more.

In my opinion it's not worth it.